src/Test/Unit/Model/Export/Catalog/ProductType/ConfigurableDataProviderTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Model\Export\Catalog\ProductType;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable as ConfigurableProductType;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Omikron\Factfinder\Api\Filter\FilterInterface;
+use Omikron\Factfinder\Model\Export\Catalog\Entity\ProductVariationFactory;
+use Omikron\Factfinder\Model\Formatter\NumberFormatter;
+use PHPUnit\Framework\TestCase;
+
+class ConfigurableDataProviderTest extends TestCase
+{
+    /** @var ConfigurableDataProvider */
+    private $configurableDataProvider;
+
+    public function test_will_return_string_on_string_value()
+    {
+        $getValueOrEmptyStringMethod = $this->invokeMethod($this->configurableDataProvider, 'getValueOrEmptyString', ['test']);
+        $this->assertEquals($getValueOrEmptyStringMethod, 'test');
+    }
+
+    public function test_will_return_empty_string_on_null_value()
+    {
+        $getValueOrEmptyStringMethod = $this->invokeMethod($this->configurableDataProvider, 'getValueOrEmptyString', [null]);
+        $this->assertEquals($getValueOrEmptyStringMethod, '');
+    }
+
+    public function test_will_return_empty_string_on_bool_value()
+    {
+        $getValueOrEmptyStringMethod = $this->invokeMethod($this->configurableDataProvider, 'getValueOrEmptyString', [false]);
+        $this->assertEquals($getValueOrEmptyStringMethod, '');
```

Comment:
This is the `assertEquals` signature
`  public static function assertEquals($expected, $actual, string $message = ''): void`

so the expected and actual values are misplaced